packages/dds/merge-tree/src/test/client.localReference.spec.ts

```diff
@@ -499,7 +503,7 @@ describe("MergeTree.Client", () => {
 		{
 			const segInfo = clients.A.getContainingSegment(9);
 			const segment = segInfo.segment;
-			assert(TextSegment.is(segment!));
+			assert(segment !== undefined && TextSegment.is(segment));
```

Comment:
TypeScript correctly requires the '!' because the type of segment is 'undefined | ISegment'.
ESLint erroneously complains that the '!' is unnecessary.
(Ditto below)